streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -237,10 +237,19 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
                 }
             }
 
-            commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+            try {
+                commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
 
-            for (final Task task : additionalTasksForCommitting) {
-                task.postCommit();
+                for (final Task task : additionalTasksForCommitting) {
+                    task.postCommit();
+                }
+            } catch (final RuntimeException e) {
```

Comment:
A `CommitFaileException` should result in a `TaskMigratedException` -- should we only catch this one? Catching a generic `RuntimeException` does seem to be a little coarse grained?

Reply:
The reason is because here we are removing the tasks before the commit block, so effectively this would be the last time we are going to see these tasks before we do a lostAll. I checked other `commitOffsetsOrTransaction` cases which are ok because there would be error handling in place or it is not inside a closing place. And if the current thread crashes for some non-recoverable exception during `commit`, these half-closed tasks are already polluting the rocksDBMetricsRecordingTrigger which later will bring down other stream threads as well.

Reply:
Ack. Thanks for clarifying.

I am wondering if we would need to cover `task.prepareCommit()` from above as well?

Reply:
One more follow up question: With incremental rebalancing and eos-beta, we might commit RUNNING tasks (that were never suspended etc) and task that got assigned to another thread/instance within the same transaction. Hence, if committing fails, we would also need to rollback those RUNNING tasks (that we track via `additionalTasksForCommitting`) -- is this rolling back ensured by upper levels or do we need to do something special here, too?

Reply:
My understanding is that the tasks in `additionalTasksForCommitting` are not wiped out during the task iterations, so if we through all the way up to the thread level, we should still have one last chance to call `handleLostAll` to clean them up.

Reply:
I agree that we would loose the reference to those tasks -- just wondering if the cleanup would be done correctly? (I believe yes, but worth to verify)